pkg/machine/env/dir.go

```diff
@@ -0,0 +1,151 @@
+package env
```

Comment:
Moving this to env feels kinda arbitrary and I do not really like it. Also makes reviewing this diff much harder.
I am not sure where exactly you import cycle happens.

Reply:
The cycle I hit was machine -> connection -> lock -> machine. Since these routines are frequently accessed from many machine  packages the chances of a cycle are high (hit this one before), and since they have no natural dependency on any other package (just the system environment) it makes sense to move them somewhere where they can be a leaf node in the graph. Since they are all computing directories/paths from the env, it IMO makes sense organizationally as well.  Although I am not sold on the name, went pack in forth between paths/env/dirs.

Reply:
(BTW in hindsight I should have split the refactor in a separate commit to make it easier to review, will try to do that more in the future)

Reply:
Yeah a separate commit would help a lot for this stuff, also including the import path.

So wouldn't it be easier to add the lock in c/common, I abstracted all the writing code into config.EditConnectionConfig() so locking there would only require 1 place which is much more desirable and safer as we cannot forget anything.

Reply:
How about I open a PR for c/common to do the locking there and then you can just leave the code where it was and only add the port logic?

Reply:
I agree the locking logic totally belongs in common for sure. Only reason I didn't put it there initially was just being up against the gun on 5.0 and trying to avoid a common pull, but certainly the way we need to do it (moving the locking to the single edit spot)

I do think the cycles on these functions will keep happening though, so at some point I think it will need to be refactored in this direction. Just a question of 5.0 or a later release.

Reply:
Ack I am not blocking over the move then. I can live with having it in env as I cannot offer a better package name for it.